### PR TITLE
convert spine errors into display in operation outcome

### PR DESCRIPTION
**Pull request by dan-jones-iw**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ all:
 
 install: install-node install-python install-hooks
 
-build: build-specification build-coordinator build-validator build-proxies
+build: build-specification build-coordinator build-proxies
```
Comment (dan-jones-iw) [inline Makefile:15]: this change was to allow me to commit changes - dont have `mvn` installed. can we make it so we dont need maven installed to commit things? i didnt make changes to the validator so doesnt seem to make much sense to rebuild it.

File: coordinator/src/services/translation/common/index.ts [hunk 2]
```diff
@@ -173,14 +177,70 @@ export function convertMomentToHl7V3DateTime(dateTime: moment.Moment): core.Time
...
+  const bodyString = message.body.toString()
+  const syncMCCI = SYNC_SPINE_RESPONSE_MCCI_REGEX.exec(bodyString)
+  const asyncMCCI = ASYNC_SPINE_RESPONSE_MCCI_REGEX.exec(bodyString)
+  let codeableConceptArray: Array<fhir.CodeableConcept> = []
+  if (syncMCCI) {
+    codeableConceptArray = translateSyncSpineResponse(syncMCCI[0])
+  } else if (asyncMCCI) {
+    codeableConceptArray = translateAsyncSpineResponse(asyncMCCI[0])
+  } else {
+    throw new Error("Spine response doesnt match")
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/common/index.ts:233]: is this the best way to handle an unexpected spine error response?

File: coordinator/src/services/translation/common/index.ts [hunk 3]
```diff
@@ -173,14 +177,70 @@ export function convertMomentToHl7V3DateTime(dateTime: moment.Moment): core.Time
...
     resourceType: "OperationOutcome",
-    issue: [{
-      code: message.statusCode <= 299 ? "informational" : "invalid",
-      severity: message.statusCode <= 299 ? "information" : "error",
-      diagnostics: message.body?.toString()
-    }]
+    issue: codeableConceptArray.map(codeableConcept => ({
+      code:"invalid",
+      severity: "error",
+      diagnostics: message.body?.toString(),
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/common/index.ts:241]: we can take this out now - do we need diagnostics? the FHIR spec says this should be populated with 'additional diagnostic information'.

File: Makefile (hunk 1 above)
Comment (mfjarvis) [reply Makefile:15]: Think this has broken the build - it's called by the pipeline as well

Comment (mfjarvis) [reply Makefile:15]: Why not install maven? It takes like 5 seconds

File: coordinator/src/services/translation/common/index.ts (hunk 3 above)
Comment (mfjarvis) [reply coordinator/src/services/translation/common/index.ts:241]: I agree we want to get rid of it eventually. I'd be inclined to leave it in for now as will help us look into issues

File: coordinator/src/services/translation/common/index.ts (hunk 2 above)
Comment (mfjarvis) [reply coordinator/src/services/translation/common/index.ts:233]: It's _a_ way :shrug:

Comment (mfjarvis) [reply coordinator/src/services/translation/common/index.ts:233]: (I think it's fine as long as we log out the content of the response somewhere, which I think Axios already does for us)

**Review by mfjarvis — APPROVED** [review]

**Pull request merged**